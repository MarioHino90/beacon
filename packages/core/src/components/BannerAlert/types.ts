import React from "react";
import { ThemedStyledProps } from "styled-components";

import { UntransitionedBannerAlertProps } from "../UntransitionedBannerAlert";

export type Intent = "success" | "warning" | "danger" | "info";

export interface BannerAlertProps
  extends UntransitionedBannerAlertProps,
    Omit<StyledBannerTransitionProps, "children"> {}

export interface StyledBannerTransitionProps
  extends React.HTMLAttributes<HTMLDivElement> {
  isOpen: boolean;
}

export interface IntentContextProviderProps {
  intent: Intent;
}

// this uses <any, any> because the extra props in the themed styled component
// do not matter, just that it is themed
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export interface IntentColorArgs extends ThemedStyledProps<any, any> {
  intent: Intent;
}
