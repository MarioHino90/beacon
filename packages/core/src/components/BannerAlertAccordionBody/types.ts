export interface BannerAlertAccordionBodyProps
  extends React.HTMLAttributes<HTMLDivElement> {}
