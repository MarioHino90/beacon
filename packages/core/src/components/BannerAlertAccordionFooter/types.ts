export interface BannerAlertAccordionFooterProps
  extends React.HTMLAttributes<HTMLDivElement> {}
