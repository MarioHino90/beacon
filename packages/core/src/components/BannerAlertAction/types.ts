import { ButtonProps } from "../Button";
import { Intent } from "../BannerAlert";

export interface BannerAlertActionProps
  extends Omit<ButtonProps, "variant" | "intent"> {
  /** * the visual intent */
  intent?: Intent;
}
