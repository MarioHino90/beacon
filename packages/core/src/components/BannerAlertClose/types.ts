import { Intent } from "../BannerAlert";

export interface BannerAlertCloseProps {
  /**
   * visual banner intents */
  intent: Intent;
  /**
   * on close callack */
  onClose: () => void;
}

export interface StyledProps {
  /**
   * visual banner intents */
  intent: Intent;
}
