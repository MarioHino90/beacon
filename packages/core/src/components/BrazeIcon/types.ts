import React from "react";

export type GenericSpotIcon<Props> = React.LazyExoticComponent<{
  (props: Props): React.ReactElement;
  displayName: string;
}>;

export interface SpotIcons {
  "add-document": GenericSpotIcon<AddDocumentProps>;
  "add-image-or-document": GenericSpotIcon<AddImageOrDocumentProps>;
  "arrow-toggle": GenericSpotIcon<ArrowToggleProps>;
  "campaign-comparison": GenericSpotIcon<CampaignComparisonProps>;
  content: GenericSpotIcon<ContentProps>;
  "email-settings": GenericSpotIcon<EmailSettingsProps>;
  "empty-table": GenericSpotIcon<AddDocumentProps>;
  "funnel-report": GenericSpotIcon<FunnelReportProps>;
  mms: GenericSpotIcon<MMSProps>;
  preview: GenericSpotIcon<PreviewProps>;
  "retention-report": GenericSpotIcon<RetentionReportProps>;
  sms: GenericSpotIcon<SMSProps>;
}

export interface BrazeIconProps extends StyledBrazeIconProps {
  /**
   * icon to be displayed to user */
  name?: keyof SpotIcons;
}

export interface StyledBrazeIconProps
  extends React.HTMLAttributes<HTMLElement> {
  /**
   * child elements to be rendered inside the component tree */
  children?: React.ReactNode | React.ReactNode[];
}

export interface AddDocumentProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface AddImageOrDocumentProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface ArrowToggleProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface CampaignComparisonProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface ContentProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface FunnelReportProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface EmailSettingsProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface PreviewProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface MMSProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface RetentionReportProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface SMSProps extends React.SVGAttributes<HTMLOrSVGElement> {}
