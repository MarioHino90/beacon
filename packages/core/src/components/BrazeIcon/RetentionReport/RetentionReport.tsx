import React from "react";
import classNames from "classnames";

import { RetentionReportProps } from "../types";

export const RetentionReport = ({
  className,
  ...props
}: RetentionReportProps): React.ReactElement => (
  <svg
    {...props}
    className={classNames(["bcl-retention-report", className])}
    width="24"
    height="24"
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g clipPath="url(#clip0)">
      <rect
        x="0.799988"
        y="2"
        width="22.425"
        height="4"
        fill="white"
        stroke="black"
        strokeLinejoin="round"
      />
      <rect
        x="0.799988"
        y="10"
        width="15.6"
        height="4"
        fill="white"
        stroke="black"
        strokeLinejoin="round"
      />
      <rect
        x="0.799988"
        y="18"
        width="7.8"
        height="4"
        fill="white"
        stroke="black"
        strokeLinejoin="round"
      />
      <g style={{ mixBlendMode: "multiply" }}>
        <rect x="0.799988" y="2" width="3.9" height="4" fill="#B5EDFF" />
      </g>
      <g style={{ mixBlendMode: "multiply" }}>
        <rect x="0.799988" y="10" width="3.9" height="4" fill="#B5EDFF" />
      </g>
      <g style={{ mixBlendMode: "multiply" }}>
        <rect x="0.799988" y="18" width="3.9" height="4" fill="#B5EDFF" />
      </g>
    </g>
    <defs>
      <clipPath id="clip0">
        <rect width="24" height="24" fill="white" />
      </clipPath>
    </defs>
  </svg>
);

RetentionReport.displayName = "RetentionReport";
