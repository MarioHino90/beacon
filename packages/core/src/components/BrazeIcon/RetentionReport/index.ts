export { RetentionReport } from "./RetentionReport";

export type { RetentionReportProps } from "../types";
