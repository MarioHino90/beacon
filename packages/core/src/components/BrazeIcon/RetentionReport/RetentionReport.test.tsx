import React from "react";
import { shallow } from "enzyme";
import faker from "@faker-js/faker";

import { RetentionReportProps } from "../types";

import { RetentionReport } from "./RetentionReport";

faker.seed(1603832397905);

describe("RetentionReport", () => {
  let props: RetentionReportProps;

  beforeEach(() => {
    props = {
      children: faker.random.words(),
      className: faker.random.word(),
      id: faker.random.word(),
    };
  });

  it("should match the snapshot", () => {
    const wrapper = shallow(<RetentionReport {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
