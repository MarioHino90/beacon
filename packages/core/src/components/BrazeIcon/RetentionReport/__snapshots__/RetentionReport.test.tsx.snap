// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetentionReport should match the snapshot 1`] = `
<svg
  className="bcl-retention-report Legacy"
  fill="none"
  height="24"
  id="matrix"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    clipPath="url(#clip0)"
  >
    <rect
      fill="white"
      height="4"
      stroke="black"
      strokeLinejoin="round"
      width="22.425"
      x="0.799988"
      y="2"
    />
    <rect
      fill="white"
      height="4"
      stroke="black"
      strokeLinejoin="round"
      width="15.6"
      x="0.799988"
      y="10"
    />
    <rect
      fill="white"
      height="4"
      stroke="black"
      strokeLinejoin="round"
      width="7.8"
      x="0.799988"
      y="18"
    />
    <g
      style={
        {
          "mixBlendMode": "multiply",
        }
      }
    >
      <rect
        fill="#B5EDFF"
        height="4"
        width="3.9"
        x="0.799988"
        y="2"
      />
    </g>
    <g
      style={
        {
          "mixBlendMode": "multiply",
        }
      }
    >
      <rect
        fill="#B5EDFF"
        height="4"
        width="3.9"
        x="0.799988"
        y="10"
      />
    </g>
    <g
      style={
        {
          "mixBlendMode": "multiply",
        }
      }
    >
      <rect
        fill="#B5EDFF"
        height="4"
        width="3.9"
        x="0.799988"
        y="18"
      />
    </g>
  </g>
  <defs>
    <clipPath
      id="clip0"
    >
      <rect
        fill="white"
        height="24"
        width="24"
      />
    </clipPath>
  </defs>
</svg>
`;
